src/data-search.js

```diff
@@ -0,0 +1,76 @@
+#! /usr/bin/env node
+
+// here we will convert the search.sh script into JS
+import { format, sub } from 'date-fns';
+import fs from 'fs';
+import { download, search, sendOutput } from './cli.js';
+
+const CATEGORY_ID = 'alzheimers-disease';
+const DATA_DIRECTORY = 'example-data';
+
+const MEDRXIV_SOURCE = 'medrxiv';
+const BIORXIV_SOURCE = 'biorxiv';
+
+const now = new Date();
+const startOffset = { days: 1 };
+const START_DATE = format(sub(now, startOffset), 'yyyy-MM-dd');
+const END_DATE = format(now, 'yyyy-MM-dd');
+
+const BIORXIV_FILE = `${DATA_DIRECTORY}/${END_DATE}_${BIORXIV_SOURCE}.json`;
+const MEDRXIV_FILE = `${DATA_DIRECTORY}/${END_DATE}_${MEDRXIV_SOURCE}.json`;
+const COMBINED_FILE = `${DATA_DIRECTORY}/${END_DATE}.json`;
+const OUTPUT_FILE = `${DATA_DIRECTORY}/${CATEGORY_ID}.json`;
+
+// Getting all latest articles from BiorXiv
+console.log(`Fetching from ${BIORXIV_SOURCE} between ${START_DATE} and ${END_DATE}`);
+fs.open(BIORXIV_FILE, 'w', function (err, file) { // consider changing the callback function if needed
+  if (err) throw err;
+  console.log('Saved!');
+});
+const bioOptions = {
+  source: BIORXIV_SOURCE,
+  output: BIORXIV_FILE
+};
+const bioData = await download(START_DATE, END_DATE, bioOptions);
+
+// Getting all latest articles from MedrXiv
+console.log(`Fetching from ${MEDRXIV_SOURCE} between ${START_DATE} and ${END_DATE}`);
+fs.open(MEDRXIV_FILE, 'w', function (err, file) {
+  if (err) throw err;
+  console.log('Saved!');
+});
+const medOptions = {
+  source: MEDRXIV_SOURCE,
+  output: MEDRXIV_FILE
+};
+const medData = await download(START_DATE, END_DATE, medOptions);
+
+// Creating a JSON with all the results, both sources combined
+console.log('Combining results...');
+fs.open(COMBINED_FILE, 'w', function (err, file) {
+  if (err) throw err;
+  console.log('Saved!');
+});
+const combinedData = {
+  ...bioData,
+  ...medData
+};
+const combinedOptions = {
+  output: COMBINED_FILE
+};
+sendOutput(combinedData, combinedOptions);
```

Comment:
I think I fixed the issue, I realized I was missing `await` in front if the async function sendOutput() here. The error doesn't come up anymore, I'm assuming what happened was the JSON file that combined the BiorXiv and MedrXiv preprint info wasn't finished populating the data before the keyword search started.